internal/source/git.go

```diff
@@ -0,0 +1,189 @@
+package source
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/nlepage/go-tarfs"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"github.com/operator-framework/rukpak/internal/git"
+	"github.com/operator-framework/rukpak/internal/util"
+)
+
+type Git struct {
+	Client          client.Client
+	KubeClient      kubernetes.Interface
+	ProvisionerName string
+	PodNamespace    string
+	UnpackImage     string
+	GitClientImage  string
+}
+
+const gitBundleUnpackContainerName = "bundle"
+
+func (g *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeGit {
+		return nil, fmt.Errorf("bundle source type %q not supported", bundle.Spec.Source.Type)
```

Comment:
Any value in surfacing that we expect this source type to be git in this error message too?